core/services/workflows/engine.go

```diff
@@ -21,98 +24,126 @@ const (
 
 type Engine struct {
 	services.StateMachine
-	logger     logger.Logger
-	registry   types.CapabilitiesRegistry
-	trigger    capabilities.TriggerCapability
-	consensus  capabilities.ConsensusCapability
-	targets    []capabilities.TargetCapability
-	workflow   *Workflow
-	callbackCh chan capabilities.CapabilityResponse
-	cancel     func()
+	logger       logger.Logger
+	registry     types.CapabilitiesRegistry
+	workflow     *workflow
+	store        *store
+	queue        *queue[stepRequest]
+	callbackCh   chan capabilities.CapabilityResponse
```

Comment:
rename to "triggerCallbackCh"?

Reply:
I've renamed it to `triggerEvents` on Henry's suggestion.